hoomd/md/TwoStepNPTMTK.cc

```diff
@@ -839,10 +891,102 @@ void TwoStepNPTMTK::advanceThermostat(unsigned int timestep)
     setIntegratorVariables(v);
     }
 
+void TwoStepNPTMTK::setCouple(const std::string& value)
+    {
+    if (value == "x")
+        {
+        m_couple = couple_x
+        }
+    else if (value == "y")
+        {
+        m_couple = couple_y
+        }
+    else if (value == "z")
+        {
+        m_couple = couple_z
+        }
+    else if (value == "xy")
+        {
+        m_couple = couple_xy
+        }
+    else if (value == "xz")
+        {
+        m_couple = couple_xz
+        }
+    else if (value == "yz")
+        {
+        m_couple = couple_yz
+        }
+    else if (value == "xyz")
+        {
+        m_couple = couple_xyz
+        }
+    }
+
+std::string TwoStepNPTMTK::getCouple()
+    {
+    std::string delimiter = "_";
+    std::string split = m_couple.substr(1, s.find(delimiter));
+    return split
+    }
+ /*   
+std::string TwoStepNPTMTK::getCouple()
+    {
+    std::string couple;
+    if (m_couple == couple_x)
+        {
+        couple = "x"
+        }
+    else if (m_couple == couple_y)
+        {
+        couple = "y"
+        }
+    else if (m_couple == couple_z)
+        {
+        couple = "z"
+        }
+    else if (m_couple == couple_xy)
+        {
+        couple = "xy"
+        }
+    else if (m_couple == couple_xz)
+        {
+        couple = "xz"
+        }
+    else if (m_couple == couple_yz)
+        {
+        couple = "yz"
+        }
+    else if (m_couple == couple_xyz)
+        {
+        couple = "xyz"
+        }
+    return couple
+    }
+*/
+
 TwoStepNPTMTK::couplingMode TwoStepNPTMTK::getRelevantCouplings()
     {
     couplingMode couple = m_couple;
 
+    // if sys_def is 2D, silently ignore any couplings that involve z
+    bool twod = m_sysdef->getNDimensions()==2;
+    if (twod) 
```

Comment:
I would make this check in `setCouple`, not `getRelevantCouplings`. We could also issue a warning to the user if they set a coupling that is ignored.

Reply:
@joaander  I'm making the change. I'm moving the checkpoint to `setCouple` and inserted warning message in the case of z coupling ignored.